apps/designer/app/designer/features/sidebar-left/panels/asset-manager/asset-thumbnail.tsx

```diff
@@ -67,15 +68,17 @@ export const AssetThumbnail = (asset: Asset | UploadingAsset) => {
         alignItems: "center",
         padding: "0 $2",
         position: "relative",
-
-        "&:hover button": {
-          opacity: 1,
-        },
       }}
+      onMouseEnter={() => setIsBeingHovered(true)}
```

Comment:
oh got why this needed to be javascrtip event handling ... Sara!

Reply:
I will check this out now

Reply:
You said css bad

Reply:
OMG NO!

Reply:
semantic props

Reply:
you talk to a component over props and tell it what to do in a semantic way, component decides what css it needs to render

Reply:
writing cascading selectors is a thing you do inside a component targeting elements you control in that component, not the child components that may have conflicting selectors, by doing so you are breaching the component boundary.

cascading like this is a thing before components age, think of components as of  custom elements shadow-root, you wouldn't be able to cascade into the component like this

Reply:
passing handlers is not breaching the  boundary, but this is generally less efficient, so not a great way either